### ISSUE TITLE
Sharing: Custom sharing service icons are shrunk to the middle of the circle

### ISSUE THREAD

**Issue by chaselivingston** [Automattic/jetpack#2451]
When creating a custom sharing button for a service not already included in Jetpack, no matter what size icon is used, it's always shrunk to the middle of the circle rather than filling up the entire circle.

Example: 
![screen shot 2015-07-28 at 09 22 24](https://cloud.githubusercontent.com/assets/831377/8932331/7b90348a-350a-11e5-8f6f-9e84961a3a88.png)

**Comment by georgestephanis** [comment]
cc: @jasmussen -- thoughts?  This may be because it's expecting the icon uploaded to just be the icon in the center, and not account for the background color and such as well?

**Comment by chaselivingston** [comment]
That makes sense, would be really cool if we could either automatically detect and set the proper background, or just have the icon fill the entire space.

**Comment by jasmussen** [comment]
The work I did on this I would mostly label "trash pickup", insofar as I only redesigned and polished and did very little on the feature side. I say that because I remember the icons that you could upload as being locked in a 16x16 maximum size. I could be misremembering or it could be different now, but because of the 16x16 size there simply wasn't resolution enough to do anything but just center them in the circle. 

There's also the default visual style of the buttons to take into account — the gray button style.

Be all that as it may — I dig the idea that a user can upload a large image, say 64x64 or 128x128 logo, that would then be scaled down and made into a circle for those custom buttons. Could be transparent background for the gray color, or they could add a fill to color it. I'm imagining CSS like this (pseudo code):

```
.custom-button {
width: 32px;
height: 32px;
}
.custom-button img {
width: 100%;
height: 100%;
border-radius: 100%;
}
```

**Comment by jeherve** [comment]
Also reported here:
https://wordpress.org/support/topic/share-button-issue-1

**Comment by csonnek** [comment]
Reported in #2296113-t

**Comment by kraftbj** [comment]
Also reported in https://wordpress.org/support/topic/added-social-sharing-icon-are-small?replies=1

**Comment by danjjohnson** [comment]
Reported in 2987879-t

**Comment by danjjohnson** [comment]
Reported in 3099144-t

**Comment by jeherve** [comment]
Also in 1101116-zen

**Comment by ivan-ottinger** [comment]
Reported on 1221202-zen.

**Comment by jeherve** [comment]
Related: #16301

**Comment by jeherve** [comment]
I believe this may have been fixed in #21075. Please re-open if the issue comes back.
